src/basic/process-util.c

```diff
@@ -136,12 +136,19 @@ int get_process_cmdline(pid_t pid, size_t max_columns, ProcessCmdlineFlags flags
         assert(line);
         assert(pid >= 0);
 
-        /* Retrieves a process' command line. Replaces non-utf8 bytes by replacement character (�). If
-         * max_columns is != -1 will return a string of the specified console width at most, abbreviated with
-         * an ellipsis. If PROCESS_CMDLINE_COMM_FALLBACK is specified in flags and the process has no command
-         * line set (the case for kernel threads), or has a command line that resolves to the empty string
-         * will return the "comm" name of the process instead. This will use at most _SC_ARG_MAX bytes of
-         * input data.
+        /* Retrieves a process' command line.
+         *
+         * If PROCESS_CMDLINE_COMM_FALLBACK is specified in flags and the process has no command line set
+         * (the case for kernel threads), or has a command line that resolves to the empty string, will
+         * return the "comm" name of the process instead. This will use at most _SC_ARG_MAX bytes of input
+         * data.
+         *
+         * There are two main modes:
+         * - if the output has limited size (max_columns is specified), returns a compact non-roundtrippable
+         *   output. Non-utf8 bytes are replaced by �. The returned string is of the specified console width
+         *   at most, abbreviated with an ellipsis.
+         * - otherwise, the output is shell-quoted and escaped, and in principle can by copy-pasted into
+         *   the terminal to execute. UTF-8 output is assumed.
```

Comment:
it sounds so strange that thus function has no way to retrieve the argv[] array 1:1 like it would be passed to main(). I have the suspicion we should hide the shell escaping logic behind some flag, and if the flag is not specified return the unmodified argv[] basically. 

We might not use it that way in our regular codepaths, but I'd still do it, and it might even be useful in tests.

Alternatively, rename the function to get_process_cmdline_shell_escaped() or so, to make very clear the escaped cmdline is the only one you'll get